WordPress/gutenberg
# Tooltip: avoid unnecessary re-renders of <select> child elements

Fix: #37856 / #38113 / #41845

## What?
This PR prevents the `Tooltip` component from re-rendering child components when the children's select list is chosen.

## Why?
Through I was investigating #41845, I found that the behavior of the `onMouseDown` event when a select list is chosen varies from browser to browser.

As mentioned in [this comment](https://github.com/facebook/react/issues/12584#issuecomment-383233455), in Firefox, `onMouseDown` is also fired when a select list is selected.

> Was just having a little poke at this. I believe @skiritsis is on the right track. There are two MouseDown events getting fired. First, a MouseDown fires when the select is initially interacted with and the browser shows the list of values. Second, a MouseDown again fires when any specific value is chosen from that modal list. If you throw a console.log into that handleMouseDown function to output the value of the select list, it will show the correct value prior to the setState() call.

This means that in Firefox, the `onMouseDown` event in `Tooltip` is also triggered when the children's select list is selected.

I'm not sure, but I would expect that the `Tooltip`'s `onMousedown` firing would cause the child component to re-render and the select element's `onChange` event is triggered with the old value.

As far as I know, this problem occurs in the following cases in Firefox:

- When changing the unit for a single direction in `BoxControl` (`BoxControl` has `UnitControl` components surrounded by a `Tooltip`.)
  - #37856
  - #41845
- When there is a `SelectControl` in the `Tooltip` component
  - #38113

## How?
I have made it so that the process is canceled only when the select list is selected in the `onMouseDown` event of the Tooltip component.
I think there is a smarter way to handle this.

## Testing Instructions

- Open Firefox.
- Insert any block.
- From the sidebar, view the item with the `BoxControl` and click "Unlink Sides".
- Confirm that the units in either direction are changed correctly.
